pkg/azure/ipam/instances.go

```diff
@@ -26,11 +27,12 @@ type AzureAPI interface {
 // InstancesManager maintains the list of instances. It must be kept up to date
 // by calling resync() regularly.
 type InstancesManager struct {
-	mutex     lock.RWMutex
-	instances *ipamTypes.InstanceMap
-	vnets     ipamTypes.VirtualNetworkMap
-	subnets   ipamTypes.SubnetMap
-	api       AzureAPI
+	mutex      lock.RWMutex
+	resyncLock lock.RWMutex
```

Comment:
I suggest to add a comment here to explain the purpose of `resyncLock` and to specify the lock ordering between that and `mutex`

Nit: might be good for readability to move `resyncLock` as the first field in the struct too, e.g:

```go
type InstancesManager struct {
	resyncLock lock.RWMutex

	mutex      lock.RWMutex
	instances  *ipamTypes.InstanceMap
	vnets      ipamTypes.VirtualNetworkMap
	subnets    ipamTypes.SubnetMap
	api        AzureAPI
}
```

to help the casual reader understand that `mutex` is meant to protect the fields below, while `resyncLock` enforces ordering between full and and incremental resyncs

Reply:
👍 Updated and applied the same change to the aws-eni and alibabacloud `InstancesManager` structs that also have a similar `resyncLock` mutex